### PR TITLE
Add `retryCount` to persistent subscription events

### PR DESCRIPTION
**Pull request by George-Payne**
- Add `PersistentSubscriptionToStreamResolvedEvent` and `PersistentSubscriptionToAllResolvedEvent` types
- Update `PersistentSubscriptionToStream` and `PersistentSubscriptionToAll` types
- Split apart event converter function types
- Add converters for persistent subscritions
- Use new types in `ReadStream`, `Subscription`, and `PersistentSubscriptionImpl`
- Use new converters in `subscribeToPersistentSubscriptionToStream` and `subscribeToPersistentSubscriptionToAll`
- Add tests for `retryCount` in `subscribeToPersistentSubscriptionToStream` and `subscribeToPersistentSubscriptionToAll`
- Update samples to show `retryCount`

### CODE REVIEW

File: src/types/events.ts [hunk 1]
```diff
@@ -195,6 +207,17 @@ export interface AllStreamResolvedEvent {
   commitPosition?: bigint;
 }
 
+/**
+ * A structure representing a single event or an resolved link event.
```
Comment (ylorph) [inline src/types/events.ts:211]: maybe add something about being a single / link for persistent subscription ?

**Review by YoEight — APPROVED** [review]

File: src/types/events.ts (hunk 1 above)
Comment (George-Payne) [inline src/types/events.ts:211]:
```suggestion
 * A structure representing a single event or a resolved link event from a persistent subscription.
```

**Review by jageall — CHANGES_REQUESTED** [review]
Does this need a breaking change / major version label? other than that looks ok to me

**Comment by George-Payne** [comment]
> Does this need a breaking change / major version label? other than that looks ok to me

Should be fully backwards compatible, `PersistentSubscriptionToStreamResolvedEvent` extends `ResolvedEvent` so can be used everywhere that a `ResolvedEvent` was previously used. 

The only signature with a breaking change is `ConvertGrpcEvent`, which is internal (The type used to describe a function that converts an event from raw GRPC to the returned object).

**Review by ylorph — APPROVED** [review]

**Pull request merged**